src/main/scala/se/lth/immun/TargetDeisotoper.scala

```diff
@@ -62,74 +167,64 @@ class TargetDeisotoper(val params:DinosaurParams) extends Timeable {
-	def extendSeed(seed:Int, inds:Seq[Int], hills:Array[Hill], z:Int):Option[IsotopePatternInds] = {
+	def extendSeed(seed:Int, hills:Array[Hill], z:Int, hillsMzs:Array[Double], hillsScanIndices:Array[(Int, Int)]):Option[IsotopePatternInds] = {
 		
-		val seedTot = hills(seed).total
-			
-		def extend2(dir:Int):Seq[Int] = {
-			var ii = inds.indexOf(seed)+dir
+		val seedCenterMz = hillsMzs(seed)
+		val seedErr2 = hills(seed).total.centerMzError * hills(seed).total.centerMzError
+		val massErrorSq = params.adv.deisoSigmas * params.adv.deisoSigmas * (
+					  2.5*seedErr2)
+		val err2 = DinoUtil.SULPHUR_SHIFT * DinoUtil.SULPHUR_SHIFT / (z*z) + massErrorSq
+		val err = Math.sqrt(err2)
+		
+		def extend2(direction:Int):Seq[Int] = {
 			var nIso = 1
-			val sHill = hills(seed)
-			val sTot = sHill.total
-			var m = sTot.centerMz + dir * nIso * DinoUtil.ISOTOPE_PATTERN_DIFF / z
-			var seedErr2 = sTot.centerMzError * sTot.centerMzError
+			var m = seedCenterMz + direction * nIso * DinoUtil.ISOTOPE_PATTERN_DIFF / z
 			var isoMissing = false
 			val isos = new ArrayBuffer[Int]
-			val alts = new ArrayBuffer[Int]
-			
 			
-			def evalAlts = {
-				if (alts.nonEmpty) {
-					val corrs = alts.map(a => (a,params.adv.deisoCorrCalc(hills(seed), hills(a))))
-					val maxCorr = corrs.maxBy(_._2)
-					alts.clear
-					if (maxCorr._2 >= params.adv.deisoCorr) {
-						isos += maxCorr._1
-						nIso += 1
-						m = seedTot.centerMz + dir * nIso * DinoUtil.ISOTOPE_PATTERN_DIFF / z
-					} else 
-						isoMissing = true
-				} else
+			while (!isoMissing) {
+			  val (minIdx, maxIdx) = DinoUtil.getMinMaxIndx(hillsMzs, m, err)
+			  var maxCorr = (-1, -1.0)
+			  for (ii <- minIdx until maxIdx) {
+	  		  if (math.min(hillsScanIndices(ii)._2, hillsScanIndices(seed)._2) > 
+	  		      math.max(hillsScanIndices(ii)._1, hillsScanIndices(seed)._1)) {
+	  		    val corr = params.adv.deisoCorrCalc(hills(seed), hills(ii))
+	  		    if (corr > maxCorr._2) {
+	  		      maxCorr = (ii, corr)
+	  		    }
+	  		  }
+        }
+				if (maxCorr._2 >= params.adv.deisoCorr) {
+					isos += maxCorr._1
+					nIso += 1
+					m = seedCenterMz + direction * nIso * DinoUtil.ISOTOPE_PATTERN_DIFF / z
+				} else 
 					isoMissing = true
 			}
-			
-			while (!isoMissing && ii < inds.length && ii >= 0) {
-				val iHill = hills(inds(ii))
-				val iTot = iHill.total
-				val mDiff = iTot.centerMz - m
-				val massErrorSq = params.adv.deisoSigmas * params.adv.deisoSigmas * (
-						seedErr2 + iTot.centerMzError * iTot.centerMzError)
-				val err2 = DinoUtil.SULPHUR_SHIFT * DinoUtil.SULPHUR_SHIFT / (z*z) + massErrorSq
-				if (mDiff * mDiff <= err2) {
-					alts += inds(ii)
-					ii += dir
-				} else if (dir*mDiff > 0) {
-					evalAlts
-				} else {
-					ii += dir
-				}
-				
-			}
-			evalAlts
 			isos
 		}
 					
-		val upMatches = extend2(1)
-		val downMatches = extend2(-1)
+	  val direction = 1
+		val upMatches = extend2(direction)
+		//val downMatches = extend2(-1)
+		val downMatches = Nil
```

Comment:
Why hard-code remove `downMatches`?

Reply:
The `downMatches` cause the reported precursor m/z to possibly be lower than the one we 'target'. I understand that this prevents you from incorrectly reporting isotopes of a different analyte, but in a matches-between-runs setting it lowers the sensitivity. If you want to keep this functionality, perhaps we should create a new flag to allow switching of the behavior here.

Reply:
That makes sense I guess, but then I would suggest making a comment like that in the code, and completely removing `downMatches` to make it clear that this is very intentional.